packages/moonstone/VirtualList/VirtualListBase.js

```diff
@@ -435,10 +451,40 @@ const VirtualListBaseFactory = (type) => {
 				const row = (index - column) % dataSize / dimensionToExtent;
 				const nextColumn = nextIndex % dimensionToExtent;
 				const nextRow = (nextIndex - nextColumn) % dataSize / dimensionToExtent;
-				const numOfItemsInPage = Math.floor((clientSize + spacing) / gridSize) * dimensionToExtent;
-				const firstFullyVisibleIndex = Math.ceil(scrollPosition / gridSize) * dimensionToExtent;
-				const isCurrentItemInView = index >= firstFullyVisibleIndex && index < firstFullyVisibleIndex + numOfItemsInPage;
-				const isNextItemInView = nextIndex >= firstFullyVisibleIndex && nextIndex < firstFullyVisibleIndex + numOfItemsInPage;
+				let
+					isNextItemInView = false,
+					isCurrentItemInView = false;
+
+				if (this.props.type === 'VariableVirtualList') {
+					const container = this.uiRefCurrent.containerRef.current;
+					const inView = [index, nextIndex].map((index) => {
+						const node = this.uiRefCurrent.containerRef.current.querySelector(`[data-index='${index}']`);
+
+						if (container && node) {
+							const containerRects = container.getBoundingClientRect();
+							const nodeRects = node.getBoundingClientRect();
+
+							if (
+								nodeRects.x >= containerRects.x &&
+								nodeRects.x + nodeRects.width <= containerRects.x + containerRects.width &&
+								nodeRects.y >= containerRects.y &&
+								nodeRects.y + nodeRects.height <= containerRects.y + containerRects.height
+							) {
+								return true;
+							}
+						}
+
+						return false;
+					});
+					isCurrentItemInView = inView[0];
+					isNextItemInView = inView[1];
+					console.log(inView);
```

Comment:
oops

Reply:
Oh~ I added while refreshing the branch. I fixed it.